x/pairing/keeper/epoch_cu.go

```diff
@@ -0,0 +1,273 @@
+	var keys []string
+	for ; iterator.Valid(); iterator.Next() {
+		key := iterator.Key()
+		// Remove the prefix to get the actual UniqueEpochSession key
+		uniqueEpochSessionKey := strings.TrimPrefix(string(key), string(types.UniqueEpochSessionKeyPrefix()))
+		keys = append(keys, uniqueEpochSessionKey)
+	}
+
+	return keys
+}
+
+// GetAllUniqueEpochSessionStore gets all the UniqueEpochSession objects from the store (used for genesis)
+func (k Keeper) GetAllUniqueEpochSessionStore(ctx sdk.Context) []types.UniqueEpochSessionGenesis {
+	info := []types.UniqueEpochSessionGenesis{}
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.UniqueEpochSessionPrefix))
+	iterator := sdk.KVStorePrefixIterator(store, []byte{}) // Get an iterator with no prefix to iterate over all keys
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		epoch, provider, chainID, project, sessionID, err := types.DecodeUniqueEpochSessionKey(string(iterator.Key()))
+		if err != nil {
+			utils.LavaFormatError("could not decode UniqueEpochSessionKey", err, utils.LogAttr("key", string(iterator.Key())))
+		}
+		info = append(info, types.UniqueEpochSessionGenesis{
+			Epoch:     epoch,
+			Provider:  provider,
+			Project:   project,
+			ChainId:   chainID,
+			SessionId: sessionID,
+		})
+	}
+
+	return info
+}
+
+/* ########## ProviderEpochCu ############ */
+
+// SetProviderEpochCu sets a ProviderEpochCu in the store
+func (k Keeper) SetProviderEpochCu(ctx sdk.Context, epoch uint64, provider string, chainID string, providerEpochCu types.ProviderEpochCu) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.ProviderEpochCuKeyPrefix())
+	b := k.cdc.MustMarshal(&providerEpochCu)
+	store.Set(types.ProviderEpochCuKey(epoch, provider, chainID), b)
+}
+
+// GetProviderEpochCu returns a ProviderEpochCu for a specific epoch and provider
+func (k Keeper) GetProviderEpochCu(ctx sdk.Context, epoch uint64, provider string, chainID string) (val types.ProviderEpochCu, found bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.ProviderEpochCuKeyPrefix())
+	b := store.Get(types.ProviderEpochCuKey(epoch, provider, chainID))
+	if b == nil {
+		return val, false
+	}
+
+	k.cdc.MustUnmarshal(b, &val)
+	return val, true
+}
+
+// RemoveProviderEpochCu removes a ProviderEpochCu from the store
+func (k Keeper) RemoveAllProviderEpochCu(ctx sdk.Context, epoch uint64) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.ProviderEpochCuKeyPrefix())
+	iterator := sdk.KVStorePrefixIterator(store, []byte(strconv.FormatUint(epoch, 10)))
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		store.Delete(iterator.Key())
+	}
+}
+
+// GetAllProviderEpochCuStore returns all the ProviderEpochCu from the store (used for genesis)
+func (k Keeper) GetAllProviderEpochCuStore(ctx sdk.Context) []types.ProviderEpochCuGenesis {
+	info := []types.ProviderEpochCuGenesis{}
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.ProviderEpochCuPrefix))
```

Comment:
use the getter for the prefix, just to be consistent

Reply:
done. see https://github.com/lavanet/lava/commit/51c70a28857299d4dab82b8907b750d1967f2bf6